commit 1f7b25ea76a925aca690da28de9d78db7ca99d0c
Author: Manoj Gupta <manojgupta@google.com>
Date:   Fri Aug 13 13:25:14 2021 -0700

    [lldb] skip host build for lldb_tblgen with LLDB_TABLEGEN_EXE set
    
    When cross compiling lldb-server, do not create a host build
    for building lldb-tblgeb when LLDB_TABLEGEN_EXE is already
    provided. This avoids an expensive and time-consuming build step
    if lldb-tblgen was already built previously for host.
    
    Reviewed By: JDevlieghere
    
    Differential Revision: https://reviews.llvm.org/D108053

diff --git a/lldb/CMakeLists.txt b/lldb/CMakeLists.txt
index 2bb05c1e220b..594c769141b4 100644
--- a/lldb/CMakeLists.txt
+++ b/lldb/CMakeLists.txt
@@ -64,7 +64,7 @@ if(LLVM_ENABLE_MODULES)
   list(APPEND LLVM_COMMON_DEPENDS intrinsics_gen)
 endif()
 
-if(CMAKE_CROSSCOMPILING AND LLDB_BUILT_STANDALONE)
+if(CMAKE_CROSSCOMPILING AND LLDB_BUILT_STANDALONE AND NOT LLDB_TABLEGEN_EXE)
   set(LLVM_USE_HOST_TOOLS ON)
   include(CrossCompile)
   if (NOT NATIVE_LLVM_DIR OR NOT NATIVE_Clang_DIR)
